src/common/mipmap_cache.c

```diff
@@ -965,7 +965,7 @@ dt_mipmap_size_t dt_mipmap_cache_get_min_mip_from_pref(const char *value)
   if(strcmp(value, "720p") == 0)   return DT_MIPMAP_3;
   if(strcmp(value, "1080p") == 0)  return DT_MIPMAP_4;
   if(strcmp(value, "WQXGA") == 0)  return DT_MIPMAP_5;
-  if(strcmp(value, "4k") == 0)     return DT_MIPMAP_6;
+  if(strcmp(value, "4K") == 0)     return DT_MIPMAP_6;
```

Comment:
Ok, i agree with uppercase. BUT - changing this as you do will lead to bad conf readings. I suggest to add `4K` here as a "synonym" fir `4k`

Reply:
There aren't many development snapshots users who already have this value set, but I agree with you, case-insensitive comparisons are cheap.

Reply:
sorry, my bad. Your first commit was good (i wrongfully remembered the other options) so let's keep the code simple and forget my "4k" :-)

Reply:
My first commit just didn't care about the few git snapshot users who could already select "4k" in the settings.

When the option for backthumbs_mipsize was written as 4k, with a lowercase letter, it could be stored in darktablerc if the user selected this option. The stored configuration value is read [here](https://github.com/darktable-org/darktable/blob/master/src/control/crawler.c#L981) and [here](https://github.com/darktable-org/darktable/blob/master/src/control/crawler.c#L981). In both places, we then call function [dt_mipmap_cache_get_min_mip_from_pref()](https://github.com/darktable-org/darktable/blob/master/src/common/mipmap_cache.c#L960) with the obtained string (ie "4k") to get the appropriate mipmap size.
Therefore, this function must compare strings case-insensitively in order to work correctly with both the old and the new spelling of the configuration value. Am I wrong somewhere?

Reply:
Nope, fully correct. From memory i thought we might also have the wrong spelling in other options also evaluating the same enums. So it's only this "just-for-a-week" problem and we might prefer the original commit...